presto-main/src/main/java/com/facebook/presto/sql/ExpressionUtils.java

```diff
@@ -322,6 +336,73 @@ public Expression rewriteLambdaExpression(LambdaExpression node, Void context, E
             {
                 return new LambdaExpression(node.getArguments(), treeRewriter.rewrite(node.getBody(), context));
             }
+
+            @Override
+            public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
```

Comment:
This looks like some uber-hack. Can comment on the purpose of that so we may think if this can be/should be replaced with some other approach.